spec/unit/resource/windows_firewall_rule_spec.rb

```diff
@@ -457,13 +491,92 @@
+        expect(provider.firewall_command("Set")).to eql("Set-NetFirewallRule -Name 'test_rule_the_second' -NewDisplayName 'some cool display name' -Description 'some other rule' -LocalAddress '192.168.40.40' -LocalPort '80' -RemoteAddress '8.8.4.4' -RemotePort '8081' -Direction 'outbound' -Protocol 'UDP' -IcmpType 'Any' -Action 'notconfigured' -Profile 'domain' -Program '%WINDIR%\\System32\\lsass.exe' -Service 'SomeService' -InterfaceType 'remoteaccess' -Enabled 'false'")
+      end
+    end
+  end
+
+  describe "#is_set_properly?" do
+    context "#TCP" do
+      it "icmp_type is nil" do
+        expect(provider.is_set_properly?(nil, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is empty" do
+        expect(provider.is_set_properly?("", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 'Any'" do
+        expect(provider.is_set_properly?("Any", "TCP")).to eql(true)
+      end
+
+      it "icmp_type is 123 as String" do
+        expect(provider.is_set_properly?("123", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 123 as Integer" do
+        expect(provider.is_set_properly?(123, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 345 as String" do
+        expect(provider.is_set_properly?("345", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is 345 as Integer" do
+        expect(provider.is_set_properly?(345, "TCP")).to eql(false)
+      end
+
+      it "icmp_type is '1:3' as code pair" do
+        expect(provider.is_set_properly?("1:3", "TCP")).to eql(false)
+      end
+
+      it "icmp_type is '123:456' as code pair out of range" do
+        expect(provider.is_set_properly?("123:456", "TCP")).to eql(false)
+      end
+    end
+
+    context "#ICMPv6" do
+      it "icmp_type is nil" do
+        expect(provider.is_set_properly?(nil, "ICMPv6")).to eql(false)
+      end
+
+      it "icmp_type is empty" do
+        expect(provider.is_set_properly?("", "ICMPv6")).to eql(false)
+      end
+
+      it "icmp_type is 'Any'" do
+        expect(provider.is_set_properly?("Any", "ICMPv6")).to eql(true)
+      end
+
+      it "icmp_type is 123 as String" do
+        expect(provider.is_set_properly?("123", "ICMPv6")).to eql(true)
+      end
+
+      it "icmp_type is 123 as Integer" do
+        expect(provider.is_set_properly?(123, "ICMPv6")).to eql(true)
+      end
+
+      it "icmp_type is 345 as String" do
+        expect(provider.is_set_properly?("345", "ICMPv6")).to eql(false)
+      end
+
+      it "icmp_type is 345 as Integer" do
+        expect(provider.is_set_properly?(345, "ICMPv6")).to eql(false)
+      end
+
+      it "icmp_type is '1:3' as code pair" do
+        expect(provider.is_set_properly?("1:3", "ICMPv6")).to eql(true)
+      end
+
+      it "icmp_type is '123:456' as code pair out of range" do
+        expect(provider.is_set_properly?("123:456", "ICMPv6")).to eql(false)
       end
```

Comment:
Is 0 a valid value? Is 254 or 255 the max allowed upper bound?

Reply:
0 is the minimum and 255 the maximum value. 
This is related to the docs (https://docs.microsoft.com/en-us/powershell/module/netsecurity/new-netfirewallrule?view=win10-ps). Search there for "-IcmpType" and you'll see the possible options.

Reply:
@phiggins I've changed the logic based on `define_resource_requirements`.
For now, I've tested the general logic for that manually, but really want to learn and get some more background information about how I could add tests for it like I did before.

Maybe we can have a discussion on Slack regarding that? Just search for pschaumburg.